### PR TITLE
Remove MobileAppUser from test app

### PR DESCRIPTION
**Pull request by aziel**
Remove MobileAppUser from test app, change how User.Id is gotten to check the NameIdentifier claim.
This will enable the E2E test app to work with the new ServerSDK.

Still need to change project refs to point to the new server sdk package once it is available.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@aziel__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (azepil). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: Runtime/Setup/dotNetApp_template.json [hunk 1]
```diff
@@ -1,48 +1,16 @@
 {
```
Comment (phvannor) [inline Runtime/Setup/dotNetApp_template.json]: can you remove these changes or generalize if the format chanegs are important

File: Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj [hunk 2]
```diff
@@ -55,35 +55,38 @@
     </Reference>
     <Reference Include="Microsoft.Azure.Mobile.Server, Version=0.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>packages\Microsoft.Azure.Mobile.Server.0.2.575\lib\net45\Microsoft.Azure.Mobile.Server.dll</HintPath>
+      <HintPath>..\..\..\azure-mobile-apps-net-server\bin\Debug\Microsoft.Azure.Mobile.Server.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Azure.Mobile.Server.Authentication, Version=0.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>packages\Microsoft.Azure.Mobile.Server.Authentication.0.2.575\lib\net45\Microsoft.Azure.Mobile.Server.Authentication.dll</HintPath>
+      <HintPath>..\..\..\azure-mobile-apps-net-server\bin\Debug\Microsoft.Azure.Mobile.Server.Authentication.dll</HintPath>
```
Comment (phvannor) [inline Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj]: can we keep the hint paths as they were as well.

File: Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj [hunk 3]
```diff
@@ -243,6 +246,10 @@
       <SubType>Designer</SubType>
     </Content>
   </ItemGroup>
+  <ItemGroup>
```
Comment (phvannor) [inline Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj]: remove

File: Runtime/DotNetE2EServerApp/Controllers/Table/AuthenticatedController.cs [hunk 4]
```diff
@@ -57,20 +68,36 @@ public class AuthenticatedController : PermissionTableControllerBase
...
 
         public override async Task<HttpResponseMessage> Delete(string id)
         {
-            MobileAppUser user = (MobileAppUser)this.User;
-            var all = (await base.GetAll()).Where(p => p.UserId == user.Id).ToArray();
+            ClaimsPrincipal user = this.User as ClaimsPrincipal;
+
+            string userId = string.Empty;
+            Claim userIdClaim = user.Claims.FirstOrDefault(c => c.Type == ClaimTypes.NameIdentifier);
+            if (userIdClaim != null)
```
Comment (phvannor) [inline Runtime/DotNetE2EServerApp/Controllers/Table/AuthenticatedController.cs]: nit: i'd prefer string userId = userIdClaim != null ? userIdClaim.Value : null;

File: Runtime/DotNetE2EServerApp/Controllers/Table/AuthenticatedController.cs [hunk 5]
```diff
@@ -19,9 +21,10 @@ public class AuthenticatedController : PermissionTableControllerBase
     {
         public override async Task<IQueryable<TestUser>> GetAll()
         {
-            MobileAppUser user = (MobileAppUser)this.User;
-            var creds = await user.GetIdentityAsync<FacebookCredentials>();
-            var all = (await base.GetAll()).Where(p => p.UserId == user.Id).ToArray();
+            ClaimsPrincipal user = this.User as ClaimsPrincipal;
+
+            var creds = await user.GetAppServiceIdentityAsync<FacebookCredentials>(this.Request);
```
Comment (phvannor) [inline Runtime/DotNetE2EServerApp/Controllers/Table/AuthenticatedController.cs]: fyi - this GetAll function I've rewritten on my side so it does something closer to what it was doing.

File: Runtime/Setup/dotNetApp_template.json (hunk 1 above)
Comment (aziel) [inline Runtime/Setup/dotNetApp_template.json]: These won't get checked in, just for my testing

File: Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj (hunk 2 above)
Comment (aziel) [inline Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj]: These also won't go in

File: Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj (hunk 3 above)
Comment (aziel) [inline Runtime/DotNetE2EServerApp/ZumoE2EServerApp.csproj]: Done

**Comment by aziel** [comment]
@phvannor  Ok, I cleaned up all the sample app and csproj changes, should be ready to go.

**Pull request merged**